src/code/z_fbdemo_circle.c

```diff
@@ -182,15 +181,19 @@ s32 TransitionCircle_IsDone(void* thisx) {
 void TransitionCircle_SetType(void* thisx, s32 type) {
     TransitionCircle* this = (TransitionCircle*)thisx;
 
-    if (type & 0x80) {
-        this->unk_14 = (type >> 5) & 0x1;
-        this->typeColor = (type >> 3) & 0x3;
-        this->speed = type & 0x1;
-        this->effect = (type >> 1) & 0x3;
+    if (type & TC_SET_PARAMS) {
+        // This bit is always going to be set under this context, this does nothing.
+        // Additionally SetType is called twice for circles, and the value will be overwritten on the
+        // second call.
+        this->direction = (type >> 5) & 0x1;
```

Comment:
It took me a while to realize these three lines of comments apply to `this->direction`. I thought the "bit" it refered to was the `TC_SET_PARAMS` bit, and "the value" refered to the four struct members

It may be clearer as
```c
        // The `(type >> 5) & 1` bit is always set in this context of being a circle type,
        // so it does not actually carry the information on the transition direction.
        // SetType is called twice for circles, the actual direction value will be set on the second call.
        this->direction = (type >> 5) & 0x1;
```